src/main/scala/org/hammerlab/guacamole/commands/VAFHistogram.scala

```diff
@@ -182,14 +181,15 @@ object VAFHistogram {
         sc.parallelize(histogramOutput).saveAsTextFile(args.output)
       } else {
         // Print histograms to standard out
-        variantAlleleHistograms.foreach(histogram =>
+        for {
+          (_, histogram) <- variantAlleleHistograms
+        } {
           histogram.toSeq.sortBy(_._1).foreach(kv => println(histogramToString(kv)))
-        )
+        }
       }
 
       if (args.cluster) {
-        val numClusters = args.numClusters
-        variantLoci.foreach(buildMixtureModel(_, numClusters))
+        buildMixtureModel(variantLoci, args.numClusters)
```

Comment:
AFAIK there's no way to apply the same trick I applied elsewhere (handling multiple independent samples in one RDD, by "keying" elements with their sample ID) to the mixture-model-building step; the mllib API we're using doesn't allow for building multiple models at once based on e.g. the key of a tuple, so this step is going to be happen in serial across samples either way.

My argument is that it's better to push that sequencing of multiple independent RDD pipelines to a different part of the stack than inside one Spark app like it's been, since Spark can't parallelize such operations at all.

Given all that, I've also added here the ability to build one model that incorporates data from multiple samples. I'm not saying that's an important use case, but more like a "might as well" after the initial step, described above, of reducing the contract here to only building one model per Spark app.

You're right that that's now how I described it in the PR message above; I'll go change that.